### ISSUE TITLE
Persist typography settings when transforming between blocks

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#34590]
## What problem does this address?
Today, if I set a Paragraph block to have a font size of “Extra Large”, then switch to a Heading block, I loose that font size attribute on the Heading block, even though I intentionally set it previously. I’d expect that the settings I opt a block into, would be carried over to the new, transformed block (if that block supports those settings).

<img width="697" alt="Screen Shot 2021-09-06 at 11 18 20 AM" src="https://user-images.githubusercontent.com/1813435/132238065-0a46a663-efb9-4b56-83cb-8c68c22ce822.png">

## What is your proposed solution?
Transforming blocks with typography settings would carry over those typography attributes.

**Comment by mtias** [comment]
Yes, this is a great point, style attributes that are supported in both the origin and target blocks could be preserved.

**Comment by Mamaduka** [comment]
I think we can pass the whole `attributes` object directly into the `createBlock` leave unsupported attribute removal to the sanitization function.

https://github.com/WordPress/gutenberg/blob/4474d01bc88d7ee4edadbf3227852e43e141fb56/packages/blocks/src/api/factory.js#L47-L51

**Comment by Humanify-nl** [comment]
I just came here in a search for issues with this.

Let me give you this specific use case:

- Set a paragraph font-size to `large` in editor.
- Transform to heading (which now inherits the large font size).

This is all fine, untill this scenario:

Theme author sets in theme.json the following:

```
	"core/heading": {
		"typography": {
			"fontSizes": []
		}
	}
```

Now the heading has inherited a large font size which **can not be removed** in the editor. 

Because there are no controls available (also pressing 'reset all' in the typography section in inspector is not removing the style.


This is not a weird use-case. Many designers set a fixed font-size for headings in a design. And to be honest, me too as a developer (giving users control over heading fontsizes, especially big ones is a nightmare for scaling designs on 5 breakpoints and managing word-wraps etc.). 

So I don't see any good reason why headings should inherit font-size when a transform is used.

What are your thoughts on this @richtabor ?

**Comment by Mamaduka** [comment]
Fixed via #37596.
